### DISCUSSION TITLE
Issue with Drawer

### DISCUSSION THREAD

**Discussion by NeilanshTriesToCode** [rsuite/rsuite#2071] [Q&A]
Hi there
### **Browser used: Chrome**
I'm trying to have 2 different Drawers on the same page. Each Drawer opens when its corresponding ``Button`` is clicked. I have set the ``placement`` for both the Drawers as ``right`` for consistency purposes. I'm managing their states using the ``useState()`` hook, so that only one Drawer is opened at a given time. 
The issue that I'm having is that the ``backdrop`` prop doesn't work for me, regardless of the value I set it to. Moreover, I the ``x`` button on the Drawer doesn't close it on clicking, and I also get an error on the console when using ``Drawer.Actions``. I have a dark background, and am trying to change the Drawer's ``backgroundColor`` through the ``style`` prop but to no avail. Has Drawer Component been discontinued? Any help would be appreciated.
Thanks

**Comment by SevenOutman** [comment]
Hi Neilansh,

> the backdrop prop doesn't work for me, regardless of the value I set it to

Could you describe how the backdrop was not working? Was it not displaying? Or was it not closing the Drawer when you clicked on it?

> I the x button on the Drawer doesn't close it on clicking

The opening state of a Drawer is _controlled_ by the `open` prop, thus the close button does not close the Drawer directly but calls the `onClose` callback. You need to set your state to false in that callback so that the Drawer gets closed.

> I also get an error on the console when using Drawer.Actions. I have a dark background, and am trying to change the Drawer's backgroundColor through the style prop but to no avail

Sounds like something's wrong. Could you provide more info about your issue like the error message you got or the code snippet that can reproduce the issue?

> Has Drawer Component been discontinued?

No, Drawer component is commonly used by us and in active maintenance.

**Reply by NeilanshTriesToCode** [reply]
I did not mention how I was using the ``Drawer`` Components previously, so apologies for that. I'm using 2 ``Drawer`` Components, both of which are set to appear from the right side. I have 2 different ``Buttons`` that allow opening the corresponding ``Drawer``. I'm using ``useState()`` to manipulate the states. For each ``Drawer``, it only opens if the ``state`` variable has been assigned to their name. For example, ``show={drawerState === 'drawerName'}`` and ``onClose={setDrawerState('')}`` to reset the state on closing a ``Drawer``.

> Sounds like something's wrong. Could you provide more info about your issue like the error message you got or the code snippet that can reproduce the issue?

This is the error I get in VS code: ``Property 'Actions' does not exist on type 'DrawerComponent'``. As for the output in Chrome, it stops displaying because it encounters the error regarding ``Drawer.Actions``. 

> Could you describe how the backdrop was not working? Was it not displaying? Or was it not closing the Drawer when you clicked on it?

My understading about the ``backdrop`` prop is that, when ``true``, it closes the ``Drawer`` if the user clicks anywhere on the remaining background, while it doesn't do it when its value is ``static`` or ``false``. In my case, I'm still not able to close it on clicking on the background when the value is set to ``true``.

> The opening state of a Drawer is controlled by the open prop, thus the close button does not close the Drawer directly but calls the onClose callback. You need to set your state to false in that callback so that the Drawer gets closed.

As for the ``open`` prop, it doesn't work for me either; the ``Drawer`` only opens for me if I'm using the ``show`` prop. I do have the ``onClose`` prop which resets the state used to maintain opening/closing  the ``Drawer``. 

On another note, I'm also not able to add my own styles (for eg. backgroundColor) to ``Drawer``, but this works for its sub-components like ``Drawer.Header`` or ''Drawer.Body``. 

Also, in one of the ``Drawer`` Components, the children Component has been made using another UI library, but that's only for one ``Drawer``, while I'm having the abovementioned problems for both ``Drawer`` Components. Do you think this could have any role?

Thanks

**Reply by SevenOutman** [reply]
> For example, `show={drawerState === 'drawerName'}` and `onClose={setDrawerState('')}` to reset the state on closing a `Drawer`.

Let me double-check this. Are you setting `onClose={setDrawerState('')}` or `onClose={() => setDrawerState('')}`? The first usage should not work because that way `onClose` is assigned the return value, not the callback.

> As for the `open` prop, it doesn't work for me either; the `Drawer` only opens for me if I'm using the `show` prop

This sounds like an impossible case. `show` prop was used in v4 and replaced by `open` in v5. Could you provide a CodeSandbox example reproducing your issue? You can use this template sandbox https://codesandbox.io/s/rsuite-template-5vq6zo2z5l 

> Property 'Actions' does not exist on type 'DrawerComponent'

This also sounds like a possible error if you were using v4. I'd suggest checking the RSuite version you are using again.

**Reply by NeilanshTriesToCode** [reply]
Yes, that's what I also thought, so I re-installed rsuite, thinking it would install the latest version, but it seems like that's not the case. In the Codebox link you sent me, it shows ``open`` and not ``show`` in Intellisense, so I believe that I still have the older version of rsuite. I would update mine, and get back to you asap  with an update. Thanks!

**Reply by NeilanshTriesToCode** [reply]
Hi, so I realized that I was using the older version of Rsuite. I'm able to access the ``open`` prop of ``Drawer``. However, the update has introduced another problem. Earlier also when I was using the older version of rsuite, the Components wouldn't show up in the output. After searching the issue up, I imported a css file for rsuite in ``App.css`` file which had then solved the issue. Since I've updated now, the css file gives me an error, and without it, the Components fail to show up on the screen. How can I resolve this issue? 
Thanks

**Reply by NeilanshTriesToCode** [reply]
``import 'rsuite/dist/styles/rsuite-default.css'`` // to fix rsuite related issues`` was the file I was importing earlier.

**Reply by NeilanshTriesToCode** [reply]
Hi again, sorry for the continuous replies, but I have managed to resolve the issue. The file to imported was ``import 'rsuite/dist/rsuite.min.css';``. Thanks a lot for the help!
